File: test/audio-capture/tone-generator.js
// Node.js tone generator using node-speaker
const Speaker = require('speaker');

// Generate a 440Hz sine wave
function generateTone(frequency, duration, sampleRate = 44100) {
    const numSamples = duration * sampleRate;
    const buffer = Buffer.alloc(numSamples * 2 * 2); // 16-bit stereo
    
    for (let i = 0; i < numSamples; i++) {
        const t = i / sampleRate;
        const sample = Math.sin(2 * Math.PI * frequency * t);
        const value = Math.floor(sample * 32767);
        
        // Write to both channels (stereo)
        buffer.writeInt16LE(value, i * 4);
        buffer.writeInt16LE(value, i * 4 + 2);
    }
    
    return buffer;
}

// Create speaker instance
const speaker = new Speaker({
    channels: 2,
    bitDepth: 16,
    sampleRate: 44100
});

console.log('440Hz Tone Generator');
console.log('Playing 440Hz tone for 3 seconds...');

// Generate and play tone
const toneBuffer = generateTone(440, 3);

speaker.on('open', () => {
    console.log('Audio device opened');
});

speaker.on('close', () => {
    console.log('Finished playing tone');
    process.exit(0);
});

// Write the tone to the speaker
speaker.write(toneBuffer);
speaker.end();

File: test/audio-capture/tone_generator.py
#!/usr/bin/env python3
"""
Simple tone generator for Windows using pyaudio
Generates a 440Hz sine wave for testing audio capture
"""

import numpy as np
import time
import sys

try:
    import pyaudio
except ImportError:
    print("Error: pyaudio not installed")
    print("Install with: pip install pyaudio")
    sys.exit(1)

def generate_tone(frequency=440, duration=3, sample_rate=44100):
    """Generate a sine wave tone"""
    samples = int(sample_rate * duration)
    wave = np.sin(2 * np.pi * frequency * np.arange(samples) / sample_rate)
    # Convert to 16-bit PCM
    wave = (wave * 32767).astype(np.int16)
    return wave

def play_tone(frequency=440, duration=3):
    """Play a tone using PyAudio"""
    print(f"=== Python Tone Generator ===")
    print(f"Frequency: {frequency} Hz")
    print(f"Duration: {duration} seconds")
    print()
    
    # Initialize PyAudio
    p = pyaudio.PyAudio()
    
    # Generate tone
    sample_rate = 44100
    tone = generate_tone(frequency, duration, sample_rate)
    
    # Open stream
    stream = p.open(
        format=pyaudio.paInt16,
        channels=1,
        rate=sample_rate,
        output=True
    )
    
    print("Playing tone...")
    
    # Play tone
    stream.write(tone.tobytes())
    
    # Cleanup
    stream.stop_stream()
    stream.close()
    p.terminate()
    
    print("Tone finished.")

if __name__ == "__main__":
    # Check if custom parameters provided
    freq = 440
    dur = 3
    
    if len(sys.argv) > 1:
        freq = int(sys.argv[1])
    if len(sys.argv) > 2:
        dur = float(sys.argv[2])
    
    play_tone(freq, dur)